src/contribute/contribution-guide.md

```diff
@@ -58,30 +58,30 @@ If interested, you can also join the other [mailing lists]({{ site.baseurl
 We use the Apache Software Foundation's
 [JIRA](https://issues.apache.org/jira/browse/BEAM) as an issue tracking and
 project management tool, as well as a way to communicate among a very diverse
-and distributed set of contributors. To be able to gather feedback, avoid
-frustration, and avoid duplicated efforts all Beam-related work should be
-tracked there.
+and distributed set of contributors. To avoid duplicated efforts,
```

Comment:
We could make this more concise - we hardly use it for project management, and most everyone knows what the point of issue tracking is. It could be "We use the ASF JIRA for issue tracking." end of para.

Reply:
(commenting here because of GitHub being better about comments on diff lines)

Above, I would actually just remove the whole bits "For larger contributions that affect multiple components....". It comes off as patronizing to me anyhow. Do software engineers working on other stuff need to be told to design their work?

And in the code / review / commit part I'd love to put a happier tone on review.

Reply:
Trimmed this down. Left "project management" as we use it for larger issues/releases as well, but cut the rest of the sentence. 

Lightened up the peer review section a bit, but mostly left for future work. The intended target audience is non-developers as well (e.g. data analysts just trying to get their job done).